ewatercycle/util.py

```diff
@@ -169,3 +171,16 @@ def data_files_from_recipe_output(recipe_output: RecipeOutput) -> Tuple[str, Dic
     # TODO simplify (recipe_output.location) when next esmvalcore release is made
     directory = str(data_files[0].filename.parent)
     return directory, forcing_files
+
+
+def to_absolute_path(input_path: str, parent: Path = None, must_exist: bool = False) -> Path:
+    """Parse input string as pathlib.Path object."""
+    pathlike = Path(input_path)
+    if parent:
+        pathlike = parent.joinpath(pathlike)
+    parsed_path = Path(pathlike).expanduser().resolve()
+    if must_exist and not parsed_path.exists():
+        raise ValueError(f"Got non-existent path {input_path}.")
+    return parsed_path
```

Comment:
Python has builtin exception [FileNotFoundError](https://docs.python.org/3/library/exceptions.html#FileNotFoundError) that is more specific than ValueError. 

As [Path.resolve(strict=True)](https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve) does exist check for you and pathlike is already a Path object, the code can be simplified to 
```suggestion
    return pathlike.expanduser().resolve(strict=must_exist)
```